### PR TITLE
Fix WARNINGs when building fastlane gemspec

### PR DESCRIPTION
**Pull request by lacostej**
Annoying warnings when building the gemspec. Clutters the logs and the mind.

```
$ gem build fastlane.gemspec
WARNING:  description and summary are identical
WARNING:  open-ended dependency on colored (>= 0) is not recommended
  use a bounded requirement, such as '~> x.y'
WARNING:  open-ended dependency on optparse (>= 0.1.1) is not recommended
  if optparse is semantically versioned, use:
    add_runtime_dependency 'optparse', '~> 0.1', '>= 0.1.1'
WARNING:  open-ended dependency on xcpretty-travis-formatter (>= 0.0.3) is not recommended
  if xcpretty-travis-formatter is semantically versioned, use:
    add_runtime_dependency 'xcpretty-travis-formatter', '~> 0.0', '>= 0.0.3'
WARNING:  See https://guides.rubygems.org/specification-reference/ for help
  Successfully built RubyGem
  Name: fastlane
  Version: 2.219.0
  File: fastlane-2.219.0.gem
```

### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.
- [x] I've added or updated relevant unit tests.

### Description

* tightened some dependencies
* added a gem summary

### CODE REVIEW

File: fastlane.gemspec [hunk 1]
```diff
@@ -75,7 +75,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency('babosa', '>= 1.0.3', '< 2.0.0') # library for creating human-friendly identifiers, aka "slugs"
   spec.add_dependency('bundler', '>= 1.12.0', '< 3.0.0') # Used for fastlane plugins
   spec.add_dependency('CFPropertyList', '>= 2.3', '< 4.0.0') # Needed to be able to read binary plist format
-  spec.add_dependency('colored') # colored terminal output
+  spec.add_dependency('colored', '~> 1.2') # colored terminal output
```
Comment (lacostej) [inline fastlane.gemspec:78]: this version was released in 2010.  So pretty safe even though in theory bumping could still be a problem?

File: fastlane.gemspec [hunk 2]
```diff
@@ -107,6 +107,6 @@ Gem::Specification.new do |spec|
   spec.add_dependency('tty-spinner', '>= 0.8.0', '< 1.0.0') # loading indicators
   spec.add_dependency('word_wrap', '~> 1.0.0') # to add line breaks for tables with long strings
   spec.add_dependency('xcodeproj', '>= 1.13.0', '< 2.0.0') # Modify Xcode projects
-  spec.add_dependency('xcpretty-travis-formatter', '>= 0.0.3')
+  spec.add_dependency('xcpretty-travis-formatter', '>= 1.0.1', '< 2.0.0')
```
Comment (lacostej) [inline fastlane.gemspec:110]: This gem is archived. Bumping isn't necessary either. Released 99% of downloads are from 1.0.0

https://rubygems.org/gems/xcpretty-travis-formatter

File: fastlane.gemspec (hunk 1 above)
Comment (rogerluan) [reply fastlane.gemspec:78]: I think this is fine

**Review by rogerluan — APPROVED** [review]
💯

**Pull request merged**